src/global/global_init.cc

```diff
@@ -454,3 +456,49 @@ int global_init_shutdown_stderr(CephContext *cct)
   return 0;
 }
 
+int global_init_preload_erasure_code(const CephContext *cct)
+{
+  const md_config_t *conf = cct->_conf;
+  string plugins = conf->osd_erasure_code_plugins;
+
+  // validate that this is a not a legacy plugin
+  list<string> plugins_list;
+  get_str_list(plugins, plugins_list);
+  for (list<string>::iterator i = plugins_list.begin();
+       i != plugins_list.end();
+       ++i) {
+	string plugin_name = *i;
+	string replacement_plugin = "";
+
+	if (plugin_name == "jerasure_generic" || 
+	    plugin_name == "jerasure_sse3" ||
+	    plugin_name == "jerasure_sse4" ||
+	    plugin_name == "jerasure_neon") {
+	  replacement_plugin = "jerasure";
+	}
+	else if (plugin_name == "shec_generic" ||
+		 plugin_name == "shec_sse3" ||
+		 plugin_name == "shec_sse4" ||
+		 plugin_name == "shec_neon") {
+	  replacement_plugin = "shec";
+	}
+
+	if (replacement_plugin != "") {
+	  derr << "WARNING: osd_erasure_code_plugins contains plugin "
```

Comment:
It would probably be better to have dout(0) instead of derr otherwise the user may be confused because it shows as if it was an error but the content of the message suggests it's only a warning.